pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -514,3 +561,21 @@ func computeClusterAPIMachineSetsFromClusterDeployment(clusterDeployment *cov1.C
 	}
 	return machineSets, nil
 }
+
+func (c *Controller) deleteFinalizer(clusterDeployment *cov1.ClusterDeployment) error {
+	clusterDeployment = clusterDeployment.DeepCopy()
+	controller.DeleteFinalizer(clusterDeployment, cov1.FinalizerRemoteMachineSets)
+	_, err := c.client.ClusteroperatorV1alpha1().ClusterDeployments(clusterDeployment.Namespace).UpdateStatus(clusterDeployment)
+	return err
+}
+
+func (c *Controller) addFinalizer(clusterDeployment *cov1.ClusterDeployment) error {
+	clusterDeployment = clusterDeployment.DeepCopy()
+	controller.AddFinalizer(clusterDeployment, cov1.FinalizerClusterDeployment)
```

Comment:
Is this supposed to be FinalizerRemoteMachineSets? If so we've got a test problem as well.

I noticed I wasn't getting the remote machine set finalizer having tested your code with a fake deployment.

Reply:
It should... I will make sure that the test is catching it